eng/packages/http-client-csharp-mgmt/emitter/test/resource-detection.test.ts

```diff
@@ -309,4 +309,406 @@ interface CurrentEmployees {
+}
+
+@armResourceOperations
+interface Employees {
+  get is ArmResourceRead<Employee>;
+  createOrUpdate is ArmResourceCreateOrReplaceAsync<Employee>;
+  update is ArmCustomPatchSync<
+    Employee,
+    Azure.ResourceManager.Foundations.ResourceUpdateModel<Employee, EmployeeProperties>
+  >;
+  delete is ArmResourceDeleteWithoutOkAsync<Employee>;
+  listByResourceGroup is ArmResourceListByParent<Employee>;
+}
+`,
+      runner
+    );
+    const context = createEmitterContext(program);
+    const sdkContext = await createCSharpSdkContext(context);
+    const root = createModel(sdkContext);
+    updateClients(root, sdkContext);
+    const employeeClient = getAllClients(root).find((c) => c.name === "Employees");
+    ok(employeeClient);
+    const employeeModel = root.models.find((m) => m.name === "Employee");
+    ok(employeeModel);
+    const departmentModel = root.models.find((m) => m.name === "Department");
+    ok(departmentModel);
+    const companyModel = root.models.find((m) => m.name === "Company");
+    ok(companyModel);
+    const employeeGetMethod = employeeClient.methods.find((m) => m.name === "get");
+    ok(employeeGetMethod);
+
+    const employeeMetadataDecorator = employeeModel.decorators?.find(
+      (d) => d.name === resourceMetadata
+    );
+    ok(employeeMetadataDecorator);
+    ok(employeeMetadataDecorator.arguments);
+    strictEqual(
+      employeeMetadataDecorator.arguments.resourceIdPattern,
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContosoProviderHub/companies/{companyName}/departments/{departmentName}/employees/{employeeName}"
+    );
+    strictEqual(
+      employeeMetadataDecorator.arguments.resourceType,
+      "Microsoft.ContosoProviderHub/companies/departments/employees"
+    );
+    strictEqual(
+      employeeMetadataDecorator.arguments.singletonResourceName,
+      undefined
+    );
+    strictEqual(
+      employeeMetadataDecorator.arguments.resourceScope,
+      "ResourceGroup"
+    );
+    strictEqual(employeeMetadataDecorator.arguments.methods.length, 5);
+    strictEqual(
+      employeeMetadataDecorator.arguments.methods[0].id,
+      employeeGetMethod.crossLanguageDefinitionId
+    );
+    strictEqual(employeeMetadataDecorator.arguments.methods[0].kind, "Get");
+    
+    const departmentMetadataDecorator = departmentModel.decorators?.find(
+      (d) => d.name === resourceMetadata
+    );
+    ok(departmentMetadataDecorator);
+    ok(departmentMetadataDecorator.arguments);
+    strictEqual(
+      departmentMetadataDecorator.arguments.parentResourceId,
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContosoProviderHub/companies/{companyName}"
+    );
+    
+    strictEqual(
+      employeeMetadataDecorator.arguments.parentResourceId,
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContosoProviderHub/companies/{companyName}/departments/{departmentName}"
```

Comment:
the tests are now looking great!
But all these 3 tests are validating the `employee` resource's parent.
Could we add some assertions for its parent (department resource) and grand parent (company resource)?
We should validate similar stuffs with what we have done for employee resource.